### PR TITLE
fix: ref selection expand after bottom line is over top line, and ref selection fill area(primary cell) and formula editor should clear when change sheet.

### PR DESCRIPTION
**Pull request by lumixraku**
Fix https://github.com/dream-num/univer-pro/issues/1451
Fix https://github.com/dream-num/univer-pro/issues/1939
Fix https://github.com/dream-num/univer-pro/issues/1699
Fix https://github.com/dream-num/univer-pro/issues/1025

close #xxx

## Pull Request Checklist

- [ ] Related tickets or issues have been linked in the PR description (or missing issue).
- [ ] [Naming convention](https://github.com/dream-num/univer/blob/dev/docs/NAMING_CONVENTION.md) is followed (**do please** check it especially when you created new plugins, commands and resources).
- [ ] Unit tests have been added for the changes (if applicable).
- [ ] Breaking changes have been documented (or no breaking changes introduced in this PR).

### CODE REVIEW

File: packages/sheets-formula/src/controllers/prompt.controller.ts [hunk 1]
```diff
@@ -410,9 +400,9 @@ export class PromptController extends Disposable {
             d.dispose();
 
             if (selections.length === 0) return;
-
+            // theme color should be set when selection coltrol is created, it's too late to set theme color at selecitonEnd(pointerup).
```
Comment (wzhudev) [inline packages/sheets-formula/src/controllers/prompt.controller.ts:403]: Spells propblem.

File: packages/sheets-formula/src/controllers/prompt.controller.ts [hunk 2]
```diff
@@ -1195,6 +1184,11 @@ export class PromptController extends Disposable {
 
     /**
      * Restore the sequenceNode generated by the lexer to the text in the editor, and set the cursor position.
+     *
+     * This function is invoked when create a seleciton(pointer down at spreadsheet) & update selection(drag move at spreadsheet)
+     *
+     * refSelectionsRenderService@_pointerDown --> refSelectionsRenderService@_selectionMoveStart$ --> _syncToEditor
```
Comment (wzhudev) [inline packages/sheets-formula/src/controllers/prompt.controller.ts:1190]: Same problem.

File: packages/sheets-formula/src/services/render-services/ref-selections.render-service.ts [hunk 3]
```diff
@@ -152,18 +155,28 @@ export class RefSelectionsRenderService extends BaseSelectionRenderService imple
     }
 
     private _initUserActionSyncListener() {
-        this.disposeWithMe(this.selectionMoveEnd$.subscribe((params) => this._updateSelections(params)));
+        // this._pointerDown --> this._selectionMoveStart$.next(this.getSelectionDataWithStyle());
```
Comment (wzhudev) [inline packages/sheets-formula/src/services/render-services/ref-selections.render-service.ts:158]: Same here.

File: packages/sheets-formula/src/services/render-services/ref-selections.render-service.ts [hunk 4]
```diff
@@ -196,6 +209,7 @@ export class RefSelectionsRenderService extends BaseSelectionRenderService imple
             const { scene } = this._context;
             const viewportMain = scene.getViewport(SHEET_VIEWPORT_KEY.VIEW_MAIN);
 
+            // change sheet ---> currentSkeleton$ change
```
Comment (wzhudev) [inline packages/sheets-formula/src/services/render-services/ref-selections.render-service.ts:212]: Same.

**Review by wzhudev — CHANGES_REQUESTED** [review]
There are code quality issues need to be addressed:

1. Do not write implementation details in JSDoc. I've explained the reasons in a comment below.
2. Pay attention to usages of capital letters, punctuations and JSDoc syntax, as the comments you write will be in the API references.
3. Remove dead code, not comment them out.

File: packages/core/src/sheets/worksheet.ts [hunk 5]
```diff
@@ -238,6 +238,17 @@ export class Worksheet {
         return null;
     }
 
+    /**
+     * get cellData
+     *
+     * All sheet CELL_CONTENT interceptor will be called in this method.
+     * this._sheetInterceptorService.intercept(INTERCEPTOR_POINT.CELL_CONTENT
+     *
+     * ex: cf.render.controller@_initViewModelInterceptor
```
[the remark this answers is not in the document]
Comment (lumixraku) [reply packages/core/src/sheets/worksheet.ts:247]: never comment where the API is called in JSDoc comments.  -->  The "ex: xxxx" in this comments?

File: packages/sheets-formula/src/services/render-services/ref-selections.render-service.ts (hunk 3 above)
Comment (lumixraku) [reply packages/sheets-formula/src/services/render-services/ref-selections.render-service.ts:158]: But this is not a JSDoc.

File: packages/sheets-ui/src/services/selection/selection-render-model.ts [hunk 6]
```diff
@@ -153,10 +151,15 @@ export class SelectionRenderModel implements IRangeWithCoord {
...
             this._rangeType = rangeType;
         }
-
         this.setCurrentCell(currentCell);
     }
 
+    /**
+     * Set primary cell.
+     *
+     * TODO @lumixraku there are 3 concepts for same thing, primary and current and highlight, primary is better cuz selectionModel is using primary.
```
Comment (wzhudev) [inline packages/sheets-ui/src/services/selection/selection-render-model.ts:160]: TODO 不写在 JSDoc 里面，放外面用 // 注释

**Review by wzhudev — APPROVED** [review]
LGTM

**Pull request merged**